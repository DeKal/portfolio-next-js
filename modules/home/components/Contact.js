import React from 'react'
import styled from 'styled-components/macro'
import { CONTACTS } from '~/modules/home/consts/pages'
import Map from '~/modules/home/components/contact/Map'
import Content from '~/modules/home/components/contact/Content'
import Header from '~/modules/home/components/contact/Header'
import Container from 'react-bootstrap/Container'
import Anchor, {
  ANCHOR_DISTANCE_TO_TOP_OF_SECTION
} from '~/modules/home/components/common/Anchor'
import SectionContainer from '~/modules/home/components/common/section/Container'

const Contact = () => (
  <SectionContainer>
    <Anchor id={CONTACTS} top={ANCHOR_DISTANCE_TO_TOP_OF_SECTION} />
    <Container>
      <Card>
        <Header />
        <Content />
      </Card>
    </Container>
    <Map />
  </SectionContainer>
)

export default Contact

const Card = styled.section`
  z-index: 50;
  width: 75%;
  margin: 0 auto;

  @media screen and (max-width: 800px) {
    width: 100%;
    float: none;
  }
`
