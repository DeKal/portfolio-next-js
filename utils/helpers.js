import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
