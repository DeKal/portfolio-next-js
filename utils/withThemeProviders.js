import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
