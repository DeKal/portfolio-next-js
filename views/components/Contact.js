import * as React from 'react'

const Contact = () => (
  <div className="contacts-section" id="contacts">
    <div className="container">
      <section className="contact-card">
        <header className="contact-card__header">
          <h2>Contact information</h2>
          <h4>My pleasure to meet you!!</h4>
        </header>
        <div className="contact-info">
          <div className="contact-info__logo">JD</div>
          <div className="contact-info__group">
            <div className="contact-info__additional">
              <p>Do you need a professional for lending a hand?</p>
              <p>Are you facing unfathomable problem?</p>
            </div>
            <div className="contact-info__vcard vcard">
              <h4 className="fn org">Phat Ho</h4>
              <p className="role title">Fullstack Developer</p>
              <p className="icon icon_location">
                <a className="adr js-anchor" href="#gmap">
                  <span className="country-name">VietNam</span>
                  ,&nbsp;
                  <span className="locality">Dist 1, Ho Chi Minh City</span>
                </a>
              </p>
              <p className="icon icon_mail">
                <a className="email" href="#contact">
                  hohuuphat22@gmail.com
                </a>
              </p>
              <p className="icon icon_phone">
                <a className="email" href="#contact">
                  +84 906323761
                </a>
              </p>
            </div>
            <ul className="contact-info__social">
              <li>
                <a
                  className="icon icon_fb"
                  rel="nofollow"
                  href="https://www.facebook.com/dekal.dev"
                >
                  I&apos;m on Facebook
                </a>
              </li>
              <li>
                <a
                  className="icon icon_in"
                  rel="nofollow"
                  href="https://www.linkedin.com/in/huu-phat-ho-1170b09b/"
                >
                  My Linkedin profile
                </a>
              </li>
            </ul>
          </div>
        </div>
      </section>
    </div>
    <div className="gmap" id="gmap"></div>
  </div>
)

export default Contact
