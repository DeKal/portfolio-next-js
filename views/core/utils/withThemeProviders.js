import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
