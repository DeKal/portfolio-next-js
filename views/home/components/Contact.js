import React from 'react'
import styled from 'styled-components/macro'
import { CONTACTS } from '~/views/home/consts/pages'
import Map from '~/views/home/components/contact/Map'
import Content from '~/views/home/components/contact/Content'
import Header from '~/views/home/components/contact/Header'
import Container from 'react-bootstrap/Container'
import Anchor from '~/views/home/components/common/Anchor'

const Contact = () => (
  <Wrapper>
    <Anchor id={CONTACTS} top={-40} />
    <Container>
      <Card>
        <Header />
        <Content />
      </Card>
    </Container>
    <Map />
  </Wrapper>
)

export default Contact

const Wrapper = styled.div`
  position: relative;
  padding-top: 100px;
`
const Card = styled.section`
  z-index: 50;
  width: 75%;
  margin: 0 auto;

  @media screen and (max-width: 800px) {
    width: 100%;
    float: none;
  }
`
