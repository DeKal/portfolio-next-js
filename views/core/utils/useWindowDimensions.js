import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
