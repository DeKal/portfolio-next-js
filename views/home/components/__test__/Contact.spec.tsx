import * as React from 'react'
import { mountWithTheme } from '~/views/core/utils/withThemeProviders'
import Contact from '~/views/home/components/Contact'

describe('Footer', () => {

  it('should render without throwing an error', function () {
    const wrap = mountWithTheme(<Contact />)
    expect(wrap).toMatchSnapshot()
  })

})