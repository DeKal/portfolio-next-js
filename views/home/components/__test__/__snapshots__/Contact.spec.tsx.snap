// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render without throwing an error 1`] = `
<Contact>
  <Contact__Wrapper>
    <div
      className="Contact__Wrapper-pivvp1-0 bNBAjy"
    >
      <Anchor
        id="Contacts"
        top={-40}
      >
        <span
          className="Anchor-sc-6rgj1q-0 fUCVuw"
          id="Contacts"
        />
      </Anchor>
      <Container
        fluid={false}
      >
        <div
          className="container"
        >
          <Contact__Card>
            <section
              className="Contact__Card-pivvp1-1 gNBLzb"
            >
              <Header>
                <Header__Container
                  className="contact-card__header"
                >
                  <header
                    className="Header__Container-gphrqw-0 fvoAft contact-card__header"
                  >
                    <h2>
                      <Translate
                        id="Contact: Header 1"
                      >
                        <WrappedTranslate
                          context={
                            Object {
                              "activeLanguage": undefined,
                              "addTranslation": [Function],
                              "addTranslationForLanguage": [Function],
                              "defaultLanguage": undefined,
                              "ignoreTranslateChildren": false,
                              "initialize": [Function],
                              "languages": Array [],
                              "renderToStaticMarkup": false,
                              "setActiveLanguage": [Function],
                              "translate": [Function],
                            }
                          }
                          id="Contact: Header 1"
                        >
                          Missing translationId: Contact: Header 1 for language: \${ languageCode }
                        </WrappedTranslate>
                      </Translate>
                    </h2>
                    <h4>
                      <Translate
                        id="Contact: Header 2"
                      >
                        <WrappedTranslate
                          context={
                            Object {
                              "activeLanguage": undefined,
                              "addTranslation": [Function],
                              "addTranslationForLanguage": [Function],
                              "defaultLanguage": undefined,
                              "ignoreTranslateChildren": false,
                              "initialize": [Function],
                              "languages": Array [],
                              "renderToStaticMarkup": false,
                              "setActiveLanguage": [Function],
                              "translate": [Function],
                            }
                          }
                          id="Contact: Header 2"
                        >
                          Missing translationId: Contact: Header 2 for language: \${ languageCode }
                        </WrappedTranslate>
                      </Translate>
                    </h4>
                  </header>
                </Header__Container>
              </Header>
              <Content>
                <Content__Container>
                  <div
                    className="Content__Container-wo3t79-0 OHnZj"
                  >
                    <Logo>
                      <Logo__Container>
                        <div
                          className="Logo__Container-sc-13alzrj-0 gCjni"
                        >
                          JD
                        </div>
                      </Logo__Container>
                    </Logo>
                    <Content__Group>
                      <div
                        className="Content__Group-wo3t79-1 cxxGGt"
                      >
                        <Description>
                          <Description__Container>
                            <div
                              className="Description__Container-sc-1wo7ggo-0 dvPShc"
                            >
                              <p>
                                <Translate
                                  id="Contact: Description 1"
                                >
                                  <WrappedTranslate
                                    context={
                                      Object {
                                        "activeLanguage": undefined,
                                        "addTranslation": [Function],
                                        "addTranslationForLanguage": [Function],
                                        "defaultLanguage": undefined,
                                        "ignoreTranslateChildren": false,
                                        "initialize": [Function],
                                        "languages": Array [],
                                        "renderToStaticMarkup": false,
                                        "setActiveLanguage": [Function],
                                        "translate": [Function],
                                      }
                                    }
                                    id="Contact: Description 1"
                                  >
                                    Missing translationId: Contact: Description 1 for language: \${ languageCode }
                                  </WrappedTranslate>
                                </Translate>
                              </p>
                              <p>
                                <Translate
                                  id="Contact: Description 2"
                                >
                                  <WrappedTranslate
                                    context={
                                      Object {
                                        "activeLanguage": undefined,
                                        "addTranslation": [Function],
                                        "addTranslationForLanguage": [Function],
                                        "defaultLanguage": undefined,
                                        "ignoreTranslateChildren": false,
                                        "initialize": [Function],
                                        "languages": Array [],
                                        "renderToStaticMarkup": false,
                                        "setActiveLanguage": [Function],
                                        "translate": [Function],
                                      }
                                    }
                                    id="Contact: Description 2"
                                  >
                                    Missing translationId: Contact: Description 2 for language: \${ languageCode }
                                  </WrappedTranslate>
                                </Translate>
                              </p>
                            </div>
                          </Description__Container>
                        </Description>
                        <Info>
                          <Info__Container>
                            <div
                              className="Info__Container-sc-1iekn4a-0 dUOZLD"
                            >
                              <Info__Name>
                                <h4
                                  className="Info__Name-sc-1iekn4a-1 kakJAu"
                                >
                                  <Translate
                                    id="Contact: Info Name"
                                  >
                                    <WrappedTranslate
                                      context={
                                        Object {
                                          "activeLanguage": undefined,
                                          "addTranslation": [Function],
                                          "addTranslationForLanguage": [Function],
                                          "defaultLanguage": undefined,
                                          "ignoreTranslateChildren": false,
                                          "initialize": [Function],
                                          "languages": Array [],
                                          "renderToStaticMarkup": false,
                                          "setActiveLanguage": [Function],
                                          "translate": [Function],
                                        }
                                      }
                                      id="Contact: Info Name"
                                    >
                                      Missing translationId: Contact: Info Name for language: \${ languageCode }
                                    </WrappedTranslate>
                                  </Translate>
                                </h4>
                              </Info__Name>
                              <Info__Role>
                                <div
                                  className="Info__Paragraph-sc-1iekn4a-2 Info__Role-sc-1iekn4a-3 jWIbpD"
                                >
                                  <Translate
                                    id="Contact: Info Role"
                                  >
                                    <WrappedTranslate
                                      context={
                                        Object {
                                          "activeLanguage": undefined,
                                          "addTranslation": [Function],
                                          "addTranslationForLanguage": [Function],
                                          "defaultLanguage": undefined,
                                          "ignoreTranslateChildren": false,
                                          "initialize": [Function],
                                          "languages": Array [],
                                          "renderToStaticMarkup": false,
                                          "setActiveLanguage": [Function],
                                          "translate": [Function],
                                        }
                                      }
                                      id="Contact: Info Role"
                                    >
                                      Missing translationId: Contact: Info Role for language: \${ languageCode }
                                    </WrappedTranslate>
                                  </Translate>
                                </div>
                              </Info__Role>
                              <Info__Location>
                                <div
                                  className="Info__Paragraph-sc-1iekn4a-2 Info__Location-sc-1iekn4a-4 jDZZpf"
                                >
                                  <Info__LocationIcon
                                    name="location"
                                  >
                                    <Icon
                                      className="Info__LocationIcon-sc-1iekn4a-11 kNqHud"
                                      name="location"
                                    >
                                      <Icon__Symbol
                                        className="Info__LocationIcon-sc-1iekn4a-11 kNqHud"
                                        name="location"
                                      >
                                        <div
                                          className="Icon__Symbol-lmunz5-0 hMIEEQ Info__LocationIcon-sc-1iekn4a-11 kNqHud"
                                          name="location"
                                        />
                                      </Icon__Symbol>
                                    </Icon>
                                  </Info__LocationIcon>
                                  <Info__Underline
                                    href="#map"
                                  >
                                    <a
                                      className="Info__Underline-sc-1iekn4a-10 iAHMdH"
                                      href="#map"
                                    >
                                      <Info__Country>
                                        <span
                                          className="Info__Inline-sc-1iekn4a-7 Info__Country-sc-1iekn4a-8 fFHDdn"
                                        >
                                          <Translate
                                            id="Contact: Info Country"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="Contact: Info Country"
                                            >
                                              Missing translationId: Contact: Info Country for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </span>
                                      </Info__Country>
                                      , 
                                      <Info__Address>
                                        <span
                                          className="Info__Inline-sc-1iekn4a-7 Info__Address-sc-1iekn4a-9 fwjGYe"
                                        >
                                          <Translate
                                            id="Contact: Info Dist"
                                          >
                                            <WrappedTranslate
                                              context={
                                                Object {
                                                  "activeLanguage": undefined,
                                                  "addTranslation": [Function],
                                                  "addTranslationForLanguage": [Function],
                                                  "defaultLanguage": undefined,
                                                  "ignoreTranslateChildren": false,
                                                  "initialize": [Function],
                                                  "languages": Array [],
                                                  "renderToStaticMarkup": false,
                                                  "setActiveLanguage": [Function],
                                                  "translate": [Function],
                                                }
                                              }
                                              id="Contact: Info Dist"
                                            >
                                              Missing translationId: Contact: Info Dist for language: \${ languageCode }
                                            </WrappedTranslate>
                                          </Translate>
                                        </span>
                                      </Info__Address>
                                    </a>
                                  </Info__Underline>
                                </div>
                              </Info__Location>
                              <Info__Email>
                                <div
                                  className="Info__Paragraph-sc-1iekn4a-2 Info__Email-sc-1iekn4a-5 kijFYz"
                                >
                                  <Info__EmailIcon
                                    name="mail"
                                  >
                                    <Icon
                                      className="Info__EmailIcon-sc-1iekn4a-12 hQoFNj"
                                      name="mail"
                                    >
                                      <Icon__Symbol
                                        className="Info__EmailIcon-sc-1iekn4a-12 hQoFNj"
                                        name="mail"
                                      >
                                        <div
                                          className="Icon__Symbol-lmunz5-0 hgCFKC Info__EmailIcon-sc-1iekn4a-12 hQoFNj"
                                          name="mail"
                                        />
                                      </Icon__Symbol>
                                    </Icon>
                                  </Info__EmailIcon>
                                  <Info__Underline>
                                    <a
                                      className="Info__Underline-sc-1iekn4a-10 iAHMdH"
                                    >
                                      hohuuphat22@gmail.com
                                    </a>
                                  </Info__Underline>
                                </div>
                              </Info__Email>
                              <Info__Phone>
                                <div
                                  className="Info__Paragraph-sc-1iekn4a-2 Info__Phone-sc-1iekn4a-6 gYChja"
                                >
                                  <Info__PhoneIcon
                                    name="phone"
                                  >
                                    <Icon
                                      className="Info__PhoneIcon-sc-1iekn4a-13 dwppIP"
                                      name="phone"
                                    >
                                      <Icon__Symbol
                                        className="Info__PhoneIcon-sc-1iekn4a-13 dwppIP"
                                        name="phone"
                                      >
                                        <div
                                          className="Icon__Symbol-lmunz5-0 eobYtY Info__PhoneIcon-sc-1iekn4a-13 dwppIP"
                                          name="phone"
                                        />
                                      </Icon__Symbol>
                                    </Icon>
                                  </Info__PhoneIcon>
                                  <Info__Underline>
                                    <a
                                      className="Info__Underline-sc-1iekn4a-10 iAHMdH"
                                    >
                                      +84 906323761
                                    </a>
                                  </Info__Underline>
                                </div>
                              </Info__Phone>
                            </div>
                          </Info__Container>
                        </Info>
                        <Socials>
                          <Socials__Container>
                            <ul
                              className="Socials__Container-sc-4xm7j6-0 gDDbeM"
                            >
                              <Socials__Item
                                data-test-id="socials-fb-link"
                                onClick={[Function]}
                              >
                                <li
                                  className="Socials__Item-sc-4xm7j6-1 enKvSp"
                                  data-test-id="socials-fb-link"
                                  onClick={[Function]}
                                >
                                  <Socials__IconWrapper
                                    name="fb"
                                  >
                                    <Icon
                                      className="Socials__IconWrapper-sc-4xm7j6-2 ikWiiM"
                                      name="fb"
                                    >
                                      <Icon__Symbol
                                        className="Socials__IconWrapper-sc-4xm7j6-2 ikWiiM"
                                        name="fb"
                                      >
                                        <div
                                          className="Icon__Symbol-lmunz5-0 kGBQMM Socials__IconWrapper-sc-4xm7j6-2 ikWiiM"
                                          name="fb"
                                        />
                                      </Icon__Symbol>
                                    </Icon>
                                  </Socials__IconWrapper>
                                </li>
                              </Socials__Item>
                              <Socials__Item
                                data-test-id="socials-linkedin-link"
                                onClick={[Function]}
                              >
                                <li
                                  className="Socials__Item-sc-4xm7j6-1 enKvSp"
                                  data-test-id="socials-linkedin-link"
                                  onClick={[Function]}
                                >
                                  <Socials__IconWrapper
                                    name="linkedin"
                                  >
                                    <Icon
                                      className="Socials__IconWrapper-sc-4xm7j6-2 ikWiiM"
                                      name="linkedin"
                                    >
                                      <Icon__Symbol
                                        className="Socials__IconWrapper-sc-4xm7j6-2 ikWiiM"
                                        name="linkedin"
                                      >
                                        <div
                                          className="Icon__Symbol-lmunz5-0 lbXhrE Socials__IconWrapper-sc-4xm7j6-2 ikWiiM"
                                          name="linkedin"
                                        />
                                      </Icon__Symbol>
                                    </Icon>
                                  </Socials__IconWrapper>
                                </li>
                              </Socials__Item>
                            </ul>
                          </Socials__Container>
                        </Socials>
                      </div>
                    </Content__Group>
                  </div>
                </Content__Container>
              </Content>
            </section>
          </Contact__Card>
        </div>
      </Container>
      <Map>
        <Map__Container
          id="map"
        >
          <div
            className="Map__Container-sc-1653heo-0 fTA-DEG"
            id="map"
          />
        </Map__Container>
      </Map>
    </div>
  </Contact__Wrapper>
</Contact>
`;
